src/utils/utils.ts

```diff
@@ -148,3 +150,27 @@ export const calculateFileSize = (sizeInBytes: number): string => {
 
   return (sizeInBytes / Math.pow(THRESHOLD, exp)).toFixed(DECIMAL_PLACES) + SEPARATOR + units[exp];
 };
+
+interface ICheckColumnsCombinations {
+  columns: Array<string>;
+  combinations: Array<Array<string>>;
+}
+
+/**
+ * Function to check if the combination requirements has been satisfied.
+ *
+ * @param columns - columns to be checked
+ * @param combinations - the combination requirements that need to be satisfied
+ *
+ */
+export const checkColumnsCombinations = ({ columns, combinations }: ICheckColumnsCombinations) => {
+  let isMismatch = false;
+  combinations.every((combination) => {
+    if (isMismatch || (combination.length >= 2 && combination.some((element) => !columns.includes(element)))) {
+      uiLogger.error(`Required combination unsatisfied: ${combinations} should be used as a combination.`);
```

Comment:
Use just singular `${combination}` to highlight which combination is failing even if it it's pointing only to first failing combination.

Reply:
done